### PR TITLE
add neuron volume and taper analysis

### PR DESCRIPTION
**Pull request by carshadi**
add neuron volume and taper analysis

### CODE REVIEW

**Comment by carshadi** [comment]
This pull request adds compartment volume analysis as well as mean burke taper vs. branch order

**Comment by tferr** [comment]
Welcome to the SNT community @carshadi!
Thanks for your contribution. This is great!

A couple of comments:
- [ ] It would be important to add references to the formulas used, and which assumptions/approximations are made on volume calculation
- [ ] Also important is to compare the results against [L-measure](http://cng.gmu.edu:8080/Lm/) (or [btmorph](https://btmorph.readthedocs.io/en/latest/readme.html)?) to ensure the accuracy of the result. The simplest way would perhaps to load a neuron (with traced diameters) from http://neuromorpho.org/, since metrics of neuromorpho cells are already online
- [ ] Add some sanity checks warning users analyzing reconstructions without/(or constant) diameter
- [ ]  `ij.gui.plot` is the only ij1 legacy library that you are using. The IJ2 alternative is [PlotService](https://github.com/maarzt/imagej-plot-service). I am not sure if the API is already public, if it is, it would be nice to include it.
- [ ] Why not include Hillman taper? 
Do you agree? Do you need snippets for the above? Also, Is there anything else you would like to be reviewed?

Thanks once more for your contribution!

File: src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py [hunk 1]
```diff
@@ -16,11 +16,12 @@
 '''
 
 import math
+from collections import defaultdict
 
 from tracing import (Path, PathAndFillManager, SimpleNeuriteTracer, SNTUI, Tree)
-from tracing.io import MLJSONLoader
+from tracing.io import MLJSONLoader, NeuroMorphoLoader
```
Comment (tferr) [inline src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py:22]: Several of these Imports have been renamed:
tracing.io.MJJSONLoader > tracing.io.MouseLightLoader
tracing.plot.TreePlot2D > tracing.viewer.Viewer2D
tracing.plot.TreePlot3D > tracing.viewer.Viewer3D
You can use the list in the script editor Templates>NeuroAnatomy>Boilerplate>Python here https://github.com/fiji/Simple_Neurite_Tracer/blob/scijava/src/main/resources/script_templates/Neuroanatomy/BoilerPlate/PY.py

File: src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 # @SNTService snt
 # @StatusService status
 # @UIService ui
-
+# @PlotService plotService
 
 '''
 file:       Analysis_Demo2.py
```
Comment (tferr) [inline src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py:12]: For completion, can you add your name to the author field? and a description of what the script does?

**Comment by tferr** [comment]
Hi @carshadi,
really sorry it took so long to come back to this. It is just that I was trying to tidy the scripting API, and wanted things to be settled before merging this. I had some minor comments on changes that need to be done so that it can run on the current API. If you don't have the time, let me know and I'll do them, I just thought you would like to do them yourself.
Also, we probably would need to change the script to the _Analysis_ folder so it is listed in the right submenu in the Script Editor, and we would need to rename it to something more descriptive. Any ideas? 

Also, is there any way to incorporate to write a jython version of that numpy code you had on the convex hull? that would be a killer addition! I guess there would be three options for it:
1) Make this script export tagged SWC nodes (like the JWS) and paste at the bottom the native python code, so that users can run it with native Python elsewhere
2) Port this script to a Python notebook using the [Scijava-jupyter-kernel](https://github.com/scijava/scijava-jupyter-kernel): This would allow using NumPy, SciPy, OpenCV, etc. To do this have a look at the examples in the [tutorials repository](https://github.com/imagej/tutorials)
3) Write a jython function from scratch that computes the convex hull so that the script becomes self-contained
Option 2 would be my preferred: It would be truly transformative to the field to have a neuroanatomy package with direct access to both the ImageJ and Python ecosystems.

**Review by tferr — CHANGES_REQUESTED** [review]

**Review by tferr — CHANGES_REQUESTED** [review]
HI @carshadi, This is nice! I added some minor requests, so that it runs on the new API

**Review by carshadi — COMMENTED** [review]
thanks @tferr , will update

File: src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py (hunk 1 above)
Comment (carshadi) [reply src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py:22]: imports have been renamed

File: src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py (hunk 2 above)
Comment (carshadi) [reply src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py:12]: done

**Comment by carshadi** [comment]
I mistakenly created a new pull request reflecting the requested changes instead of updating to this one. Possibly because I did not have GitKraken setup when creating the initial PR.
New pull request here https://github.com/fiji/Simple_Neurite_Tracer/pull/42

**Comment by tferr** [comment]
No problem! Thanks a lot! For future ref, this was superseded by #42 that meanwhile has been merged

**Pull request closed without merging**